packages/journey-manager/src/index.ts

```diff
@@ -209,14 +210,48 @@ function filterMap<X, Y>(
   }, []);
 }
 
+const resolveTriggers = async (
+  config: Config,
+  triggers?: Triggers,
+): Promise<Triggers> => {
+  if (triggers != null) {
+    return triggers;
+  }
+  const baseUrl =
+    config.dev ?? false
+      ? "https://triggers.dev.nlx.ai"
+      : "https://triggers.nlx.ai";
+  const triggersFromCdnRequest = await fetch(
+    `${baseUrl}/${config.workspaceId}/${config.journeyId}.json`,
+  );
+  const triggersFromCdn = await triggersFromCdnRequest.json();
+  return triggersFromCdn;
+};
+
+// eslint-disable-next-line @typescript-eslint/promise-function-async
+const waitUntilDomContentLoaded = (): Promise<unknown> => {
+  return new Promise((resolve) => {
+    window.addEventListener("DOMContentLoaded", (event) => {
+      resolve(event);
+    });
+  });
```

Comment:
See https://developer.mozilla.org/en-US/docs/Web/API/Document/DOMContentLoaded_event#checking_whether_loading_is_already_complete.

```suggestion
  if (document.readyState === "loading") {
    return new Promise((resolve) => {
      window.addEventListener("DOMContentLoaded", (event) => {
        resolve();
      });
    });
  } else {
    return Promise.resolve();
  }
```

Reply:
Oh right on, thank you 🙏🙏🙏